components/brave_ads/browser/ads_notification_handler.cc

```diff
@@ -120,4 +157,47 @@ const void* AdsNotificationHandler::UserDataKey() {
   return &kUserDataKey;
 }
 
+void AdsNotificationHandler::OperationCompleted(
+    const std::string& notification_id) {
+
+  auto iter = pending_callbacks_.find(notification_id);
+  std::move(iter->second).Run();
```

Comment:
Seems like we should check `iter` here before accessing it (or at least `DCHECK` if we're confident that it's impossible).

Reply:
Fixed